### PR TITLE
Remove branch before mirroring

### PR DESCRIPTION
**Pull request by RobertGlein**
Remove set -u to allow a check of the new variable.
Check if the new variable exists and is true.
Check if the branch exists and delete it remotely.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/bin/sh
 
 # Error handling
-set -euo pipefail
+set -eo pipefail
 # set -e : Instructs bash to immediately exit if any command has a non-zero exit status.
-# set -u : Reference to any variable not previously defined - with the exceptions of $* and $@ - is an error
+# (not active) set -u : Reference to any variable not previously defined - with the exceptions of $* and $@ - is an error
```
Comment (aperloff) [inline entrypoint.sh:6]: Did you remove `-u` because we don't always want `REMOVE_BRANCH` to be set?

File: entrypoint.sh [hunk 2]
```diff
@@ -18,6 +18,16 @@ sh -c "git config --global credential.username $GITLAB_USERNAME"
 sh -c "git config --global core.askPass /cred-helper.sh"
 sh -c "git config --global credential.helper cache"
 sh -c "git remote add mirror $*"
+if [[ -n "${REMOVE_BRANCH}" ]]; then # Check if variable exists 
+   if [[ "${REMOVE_BRANCH}" == "true" ]]; then
```
Comment (aperloff) [inline entrypoint.sh]: Why put these on two lines? Why not put them on a single line like:
```
if [[ -n "${REMOVE_BRANCH}" ]] && [[ "${REMOVE_BRANCH}" == "true" ]]; then
```

File: entrypoint.sh [hunk 3]
```diff
@@ -18,6 +18,16 @@ sh -c "git config --global credential.username $GITLAB_USERNAME"
 sh -c "git config --global core.askPass /cred-helper.sh"
 sh -c "git config --global credential.helper cache"
 sh -c "git remote add mirror $*"
+if [[ -n "${REMOVE_BRANCH}" ]]; then # Check if variable exists 
+   if [[ "${REMOVE_BRANCH}" == "true" ]]; then 
+      # If branch exists; w/o sh -c ""
+      branchExists=$(git ls-remote $(git remote get-url --push mirror) ${CHECKOUT_BRANCH:-$DEFAULT_GITHUB_REF} | wc -l)
+      if [[ ${branchExists} == 1 ]]; then
```
Comment (aperloff) [inline entrypoint.sh]: In bash, isn't it better to quote both the variable size and the comparison side of a binary operator?

File: entrypoint.sh [hunk 4]
```diff
@@ -18,6 +18,16 @@ sh -c "git config --global credential.username $GITLAB_USERNAME"
...
+         sh -c "echo removing $branch branch at $(git remote get-url --push mirror)"
```
Comment (aperloff) [inline entrypoint.sh]: Does this work? Seems like the message should be quoted somehow. Is the echo just gobbling up everything that comes after it? I haven't tested this, but I would think this line should be more like:
```
sh -c "echo \"removing the branch ${branch} at $(git remote get-url --push mirror)\""
```

Also, is the `sh -c` really necessary? Aren't you already in a bash shell?

File: entrypoint.sh [hunk 5]
```diff
@@ -18,6 +18,16 @@ sh -c "git config --global credential.username $GITLAB_USERNAME"
 sh -c "git config --global core.askPass /cred-helper.sh"
 sh -c "git config --global credential.helper cache"
 sh -c "git remote add mirror $*"
+if [[ -n "${REMOVE_BRANCH}" ]]; then # Check if variable exists 
+   if [[ "${REMOVE_BRANCH}" == "true" ]]; then 
+      # If branch exists; w/o sh -c ""
+      branchExists=$(git ls-remote $(git remote get-url --push mirror) ${CHECKOUT_BRANCH:-$DEFAULT_GITHUB_REF} | wc -l)
+      if [[ ${branchExists} == 1 ]]; then
+         sh -c "echo removing $branch branch at $(git remote get-url --push mirror)"		
+         sh -c "git push mirror --delete $branch"
```
Comment (aperloff) [inline entrypoint.sh]: Again, is the `sh -c` really necessary?

File: entrypoint.sh (hunk 2 above)
Comment (RobertGlein) [reply entrypoint.sh]: This is a matter of taste but I am fine with changing it.

File: entrypoint.sh (hunk 3 above)
Comment (RobertGlein) [reply entrypoint.sh]: I guess. I will change it

File: entrypoint.sh (hunk 2 above)
Comment (aperloff) [reply entrypoint.sh]: Since there is no `else` statement, it just seems like it should be combined.

File: entrypoint.sh (hunk 4 above)
Comment (RobertGlein) [reply entrypoint.sh]: It works but it is cleaner your way
I will remove the sh -c

File: entrypoint.sh (hunk 5 above)
Comment (RobertGlein) [reply entrypoint.sh]: I will remove the sh -c

File: entrypoint.sh (hunk 1 above)
Comment (RobertGlein) [reply entrypoint.sh:6]: Yes, to have it more generic. I do not know if it is necessary for your other projects.

Comment (aperloff) [reply entrypoint.sh:6]: This is fine with me.

File: entrypoint.sh [hunk 6]
```diff
@@ -18,6 +18,14 @@ sh -c "git config --global credential.username $GITLAB_USERNAME"
 sh -c "git config --global core.askPass /cred-helper.sh"
 sh -c "git config --global credential.helper cache"
 sh -c "git remote add mirror $*"
+if [[ -n "${REMOVE_BRANCH}" ]] && [[ "${REMOVE_BRANCH}" == "true" ]]; then # Check if variable exists and is true
+   # If branch exists; w/o sh -c ""
```
Comment (aperloff) [inline entrypoint.sh]: Please remove the part of this comment about sh: `; w/o sh -c ""`

File: entrypoint.sh [hunk 7]
```diff
@@ -18,6 +18,14 @@ sh -c "git config --global credential.username $GITLAB_USERNAME"
...
+   branchExists=$(git ls-remote $(git remote get-url --push mirror) ${CHECKOUT_BRANCH:-$DEFAULT_GITHUB_REF} | wc -l)
+   if [[ "${branchExists}" == "1" ]]; then
+      echo "removing the branch ${branch} at $(git remote get-url --push mirror)"
```
Comment (aperloff) [inline entrypoint.sh]: Can you just flip the `branch` and `${branch}` parts. Sorry, I missed that there was a similar line at the end of the file.

**Comment by aperloff** [comment]
Please test this again after making the changes.

**Comment by RobertGlein** [comment]
Test successful: https://github.com/cms-tracklet/firmware-hls/runs/644488375?check_suite_focus=true#step:7:20

**Pull request merged**